packages/design-system/src/components/Status/Status.tsx

```diff
@@ -1,37 +1,42 @@
 import React from 'react';
-import { IconName } from '@talend/icons';
-import Tooltip from '../Tooltip';
-import { Icon } from '../Icon/Icon';
-import Loading from '../Loading';
-
-import * as S from './Status.style';
-
-export type StatusProps = React.PropsWithChildren<any> & {
-	icon?: IconName;
-	inProgress?: boolean;
-	hideText?: boolean;
-	children: string;
-};
-
-const Status = React.forwardRef(
-	(
-		{ children, icon, inProgress, hideText, ...rest }: StatusProps,
-		ref: React.Ref<HTMLSpanElement>,
-	) => {
-		const text = <span className="status__text">{children}</span>;
-		const picto = (
-			<span className="status__icon" aria-hidden>
-				{!inProgress && icon && <Icon name={icon} />}
-				{inProgress && <Loading />}
-			</span>
-		);
-
-		return (
-			<S.Status ref={ref} {...rest}>
-				{hideText ? <Tooltip title={children}>{picto}</Tooltip> : [picto, text]}
-			</S.Status>
-		);
-	},
+import StatusFailed, { StatusFailedProps } from './variations/StatusFailed';
+import StatusWarning, { StatusWarningProps } from './variations/StatusWarning';
+import StatusSuccessful, { StatusSuccessfulProps } from './variations/StatusSuccessful';
+import StatusInProgress, { StatusInProgressProps } from './variations/StatusInProgress';
+import StatusCanceled, { StatusCanceledProps } from './variations/StatusCanceled';
+import { possibleVariants } from './Primitive/StatusPrimitive';
+
+type StatusProps = {
+	variant: typeof possibleVariants[number];
+} & (
+	| StatusFailedProps
+	| StatusWarningProps
+	| StatusSuccessfulProps
+	| StatusInProgressProps
+	| StatusCanceledProps
 );
 
+const Status = React.forwardRef((props: StatusProps, ref: React.Ref<any>) => {
+	const { variant, ...rest } = props;
+	switch (variant) {
+		case 'failed':
```

Comment:
We may use `enum` instead?
https://github.com/Talend/ui/blob/master/packages/design-tokens/src/types.ts#L1-L13

Reply:
I don't really see the benefit?

Reply:
Avoid typo, not a huge fan of plain text